src/Features/LanguageServer/Protocol/Handler/Rename/PrepareRenameHandler.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Rename;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler;
+
+[ExportCSharpVisualBasicStatelessLspService(typeof(PrepareRenameHandler)), Shared]
+[Method(Methods.TextDocumentPrepareRenameName)]
+internal class PrepareRenameHandler : ILspServiceDocumentRequestHandler<PrepareRenameParams, DefaultBehaviorPrepareRename?>
+{
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public PrepareRenameHandler()
+    {
+    }
+
+    public bool MutatesSolutionState => false;
+    public bool RequiresLSPSolution => true;
+
+    public TextDocumentIdentifier GetTextDocumentIdentifier(PrepareRenameParams request) => request.TextDocument;
```

Comment:
```suggestion
    public TextDocumentIdentifier GetTextDocumentIdentifier(PrepareRenameParams request)
        => request.TextDocument;
```

nit: i really prefer methods (even simple ones) having the body on the next line.  makes it immediately clear to me it's got a body.

Reply:
This rule may be worth a new analyzer in Roslyn PrivateAnalyzers project.